partitioned-heat-conduction/fenics/utils/utils.py

```diff
@@ -13,20 +13,13 @@ def b_splines(precice, degree, dt):
     '''
     # use equidistant samples
     # apparently you need 2k+2 samples for degree k
-    x_dist = float(dt) / (2 * degree + 2)
-    nodes = []
+    nodes = np.linspace(0,dt,2*3+3)  # <-- But these are 2k+3 samples
```

Comment:
This contradicts the statement above. Which one is true?

Reply:
The comment is a bit imprecise. You need at least 2k+2 samples.
I will check, if you can reduce the samples to at least 2k, as I changed this code segment here:
[https://github.com/NiklasVin/tutorials/commit/00007d7be9a4d8334089c2f91b3a25e49ed6b0ec#diff-0b22c9acd2661e0e0fe1d0350fdd7736431cd134ab6f1e5b088d90b0f8c7b2b8L34-R35]()
If you used the constructor directly instead of using the return values of `splrep`, the constructor had for some reason a problem less than 2k+2 samples were given. Maybe an implementation error from my side, but I am on it :)